### PR TITLE
Reduce redundant code

### PR DESCRIPTION
**Pull request by roclark**

### CODE REVIEW

File: sources/procfs.go [hunk 1]
```diff
@@ -376,60 +347,31 @@ func getJobStatsByOperation(jobBlock string, jobID string, operation string) (me
...
-	pattern := "(?ms:job_id:.*?(-|\\z))"
-	jobRegex, err := regexp.Compile(pattern)
-	if err != nil {
-		return nil, err
-	}
-
-	jobs := jobRegex.FindAllString(jobStats, -1)
-	if len(jobs) == 0 {
+	jobs := regexCaptureStrings("(?ms:job_id:.*?(-|\\z))", jobStats)
+	if len(jobs) < 0 {
```
Comment (joehandzik) [inline sources/procfs.go]: You're no longer returning nil, nil in cases where len(jobs) is equal to zero. Is that intentional?

Comment (roclark) [inline sources/procfs.go]: Great catch, this was supposed to be
```
if len(jobs) < 1 {
```

**Comment by joehandzik** [comment]
@roclark I don't see any other issues. My only concern is sort of an overarching concern with the whole refactor...is it an issue that there are very few cases where you're returning an error struct (almost everything is nil, nil)? If that's intentional and you don't think it'll cause problems, feel free to merge when you're ready.

**Comment by roclark** [comment]
Due to the conversion of `regexp.Compile` to `regexp.MustCompile`, we shouldn't be getting any errors thrown by this code.

**Pull request merged**